tests/e2e/specs/backend/active-filters.test.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * External dependencies
+ */
+import {
+	insertBlock,
+	getEditedPostContent,
+	getAllBlocks,
+	switchUserToAdmin,
+	openDocumentSettingsSidebar,
+} from '@wordpress/e2e-test-utils';
+
+import {
+	visitBlockPage,
+	clearAndFillInput,
+} from '@woocommerce/blocks-test-utils';
+import { insertBlockDontWaitForInsertClose } from '../../utils';
+
+const block = {
+	name: 'Active Product Filters',
+	slug: 'woocommerce/active-filters',
+	class: '.wc-block-active-filters',
+};
+
+if ( process.env.WP_VERSION < 5.3 )
+	// eslint-disable-next-line jest/no-focused-tests
+	test.only( 'skipping all other things', () => {} );
+
+describe( `${ block.name } Block`, () => {
+	beforeAll( async () => {
+		await switchUserToAdmin();
+		await visitBlockPage( `${ block.name } Block` );
+	} );
+
+	it( 'can only be inserted once', async () => {
+		await insertBlockDontWaitForInsertClose( block.name );
+		expect( await getAllBlocks() ).toHaveLength( 1 );
+	} );
+
+	it( 'renders without crashing', async () => {
+		await expect( page ).toRenderBlock( block );
+	} );
+
+	it( 'allows title can be manipulated', async () => {
+		await openDocumentSettingsSidebar();
+		await clearAndFillInput(
+			'.wp-block[data-type="woocommerce/active-filters"] textarea.wc-block-editor-components-title',
```

Comment:
I love that we now have the pptr-testing-library that allows us to search for content from a user's point of view. With Testing library I know that it's encouraged to search the page for texts the user sees, so I wonder if we could drop searching by this selector and use 

`$document.findByPlaceholderText('Active filters')`

Reply:
Hmm, I agree it would be _better_, but I'm not confident this is a good time to start using that in place of what we're doing now with selectors. My reasoning is: if we go ahead with rewriting the `clearAndFillInput` method, we will need to add some kind of fallback for when it's not possible to select something by these "text" values from `pptr-testing-library`. This then means we're juggling two things, puppeteer `elementHandles` and dom-testing-library's `HTMLElement`. Also, all tests using this will need to be rewritten to keep track of the current text. This is because some inputs don't have placeholders, or labels, see Filter by Price title, it's a `textarea` with no label or placeholder, so we'll need to know the text beforehand, then keep track of the the text it changes to.

I apologise if I've misunderstood the way nodes/elements are returned by both of these libraries.

On top of this, based on what I read here (https://testing-library.com/docs/example-input-event/) we wouldn't be able to simulate typing into an input, which is sometimes required, because changing the `.value` property of a node isn't enough, in the case of the `Filter Products by Price` title, setting `.value` doesn't persist the new title, it has to be typed "manually".

Reply:
Maybe I got a bit carried away because I am a huge fan of the [Testing library's principles](https://testing-library.com/docs/), moving away from testing implementation details (aka search for nested selectors) and, instead, test using what the user sees (texts, labels, etc).

It wasn't fully clear to me why `pptr-testing-library` was added. You mentioned making use of its queries and I thought that would be the default way from now on. I think it would be helpful to explain, maybe in a P2:
- what are the reasons behind adding it
- when this library should be used
- how do we see the adoption phase

I did not know about the limitations around typing so thanks for clearing those out. We can keep `clearAndFillInput` as it is.

Reply:
Yup, it was brought in because I based my work here on Nadir's PR - https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/2756

I liked the idea that we didn't have to go and find controls by their css selector, and could use more "user-based" selectors, so I wanted to bring it over to this PR too. In places that selectors had already been used, I wasn't sure if I should go back and rewrite them. I can certainly do that, though.